man/man1/indent.1
.\" @(#)indent.1 1.1 92/07/30 SMI;.
.TH INDENT 1 "9 September 1987"
.SH NAME
indent \- indent and format a C program source file
.SH SYNOPSIS
.in +\w'\fBindent \fR'u
.ti -\w'\fBindent \fR'u
.B indent 
.I input-file
[
.I output-file
] [
[\ \fB\-bap\fR\ |\ \fB\-nbap\fR\ ]
[\ \fB\-bacc\fR\ |\ \fB\-nbacc\fR\ ]
[\ \fB\-bad\fR\ |\ \fB\-nbad\fR\ ]
[\ \fB\-bbb\fR\ |\ \fB\-nbbb\fR\ ]
[\ \fB\-bc\fR\ |\ \fB\-nbc\fR\ ]
[\ \fB\-bl\fR\ ]
[\ \fB\-br\fR\ ]
[\ \fB\-bs\fR\ |\ \fB\-nbs\fR\ ]
[\ \fB\-c\fIn\fR\ ]
[\ \fB\-cd\fIn\fR\ ]
[\ \fB\-cdb\fR\ |\ \fB\-ncdb\fR\ ]
[\ \fB\-ce\fR\ |\ \fB\-nce\fR\ ]
[\ \fB\-ci\fIn\fR\ ]
[\ \fB\-cli\fIn\fR\ ]
[\ \fB\-d\fIn\fR\ ]
[\ \fB\-di\fIn\fR\ ]
[\ \fB\-eei\fR\ |\ \fB\-neei\fR\ ]
[\ \fB\-fc1\fR\ |\ \fB\-nfc1\fR\ ]
[\ \fB\-i\fIn\fR\ ]
[\ \fB\-ip\fR\ |\ \fB\-nip\fR\ ]
[\ \fB\-l\fIn\fR\ ]
[\ \fB\-lc\fIn\fR\ ]
[\ \fB\-lp\fR\ |\ \fB\-nlp\fR\ ]
[\ \fB\-pcs\fR\ |\ \fB\-npcs\fR\ ]
[\ \fB\-npro\fR\ ]
[\ \fB\-psl\fR\ |\ \fB\-npsl\fR\ ]
[\ \fB\-sc\fR\ |\ \fB\-nsc\fR\ ]
[\ \fB\-sob\fR\ |\ \fB\-nsob\fR\ ]
[\ \fB\-st\fR\ ]
[\ \fB\-troff\fR\ ]
[\ \fB\-v\fR\ |\ \fB\-nv\fR\ ]
.SH DESCRIPTION
.IX  indent  ""  "\fLindent\fP \(em format C source"
.IX  "programming tools"  "indent"  ""  "\fLindent\fP \(em format C source"
.IX  "languages"  "indent"  ""  "\fLindent\fP \(em format C source"
.IX  "C programming language"  "indent"  ""  "\fLindent\fP \(em format C source"
.IX  "pretty printer"  "indent"  ""  "\fLindent\fP \(em format C source"
.IX  "format C programs" "" "format C programs \(em \fLindent\fP"
.IX  "code formatter"  "indent"  ""  "\fLindent\fP \(em format C source"
.\".IX  "cb"  "indent"  "\fLcb\fP"  "try \fLindent\fP \(em format C source"
.LP
.B indent
is a C program formatter.  It reformats the C
program in the
.I input-file
according to the switches.  The switches which can
be specified are described below.
They may appear before or after the file
names.
.LP
Note: if you only specify an
.IR input-file ,
the formatting is
done \*(lqin-place\*(rq, that is, the formatted
file is written back into
.I input-file
and a backup copy of
.I input-file
is written in the current directory.  If
.I input-file
is named
.BR /blah/blah/file ,
the backup file is named
.BR file.\|\s-1BAK\s0 .
.LP
If
.I output-file
is specified,
.B indent
checks to make sure it is different from
.IR input-file .
.SH OPTIONS
.LP
The options listed below control the formatting style imposed by
.BR indent .
.TP 15
.BR \-bap , \-nbap
If
.B \-bap
is specified, a blank line is forced after
every procedure body.  Default:
.B \-nbap.
.TP
.BR \-bacc , \-nbacc
If
.B \-bacc
is specified, a blank line is forced around every conditional
compilation block.  That is. in front of every
.B #ifdef
and after every
.BR #endif .
Other blanklines surrounding these will be swallowed.
Default: 
.B \-nbacc.
.TP
.BR \-bad , \-nbad
If
.B \-bad
is specified, a blank line is forced after every block of
declarations.  Default: 
.B \-nbad.
.TP
.BR \-bbb , \-nbbb
If
.B \-bbb
is specified, a blank line is forced before
every block comment.  Default:
.B \-nbbb.
.TP
.BR \-bc , \-nbc
If
.B \-bc
is specified, then a
.SM NEWLINE
is forced after each comma in a declaration.
.B \-nbc
turns off this option.  The default is
.BR \-bc .
.TP
.BR \-br , \-bl
Specifying
.B \-bl
lines up compound statements like this:
.ne 4
.RS
.RS
.nf
.ft B
if (\|.\|.\|.)
{
.ft R
.RS
.RS
.RS
.B code
.RE
.RE
.RE
.B }
.fi
.RE
.RE
.IP
Specifying
.B \-br
(the default) makes them look like this:
.ne 3
.RS
.RS
.sp .5
.nf
.ft B
if (...) {
.ft R
.RS
.RS
.RS
.B code
.RE
.RE
.RE
.B }
.fi
.RE
.RE
.LP
.TP
.BR \-bs , \-nbs
Enable (disable) the forcing of a blank after
.BR sizeof .
Some people believe that
.B sizeof
should appear as though it were a procedure call
(\fB\-nbs\fR,
the default) and some people believe that since
.B sizeof
is an operator, it should always be treated
that way and should always have a
blank after it.
.TP
.BI \-c n
The column in which comments on code start.  The default is 33. 
.TP
.BI \-cd n
The column in which comments on declarations start.  The default
is for these comments to start in the same column as those on code.
.br
.ne 5
.TP
.BI \-cdb , \-ncdb
Enable (disable) the placement of comment
delimiters on blank lines.  With
this option enabled, comments look like this:
.ne 3
.RS
.nf
.ft B
/*
* this is a comment
*/
.ft R
.fi
.RE
.LP
Rather than like this:
.IP
.B /* this is a comment */
.LP
This only affects block comments, not comments
to the right of code. The default is
.B \-cdb .
.TP
.BI \-ce , \-nce
Enables (disables) forcing
.BR else 's
to cuddle up to the immediately preceding
.RB ` } '.
The default is
.BR \-ce .
.TP
.BI \-ci n
Sets the continuation indent to be
.IR n .
Continuation
lines will be indented that far from the
beginning of the first line of the
statement.  Parenthesized expressions have extra indentation added to
indicate the nesting, unless
.B \-lp
is in effect.
.B \-ci
defaults to the same value
as
.BR \-i .
.TP
.BI \-cli n
Cause case labels to be indented
.I n
tab stops to the right of the containing
.B switch
statement.
.B -cli0.5
causes case labels to be indented half a tab stop.
The
default is
.B \-cli0 .
.TP
.BI \-d n
Control the placement of comments which are not to the
right of code.  The default
.B \-d1
means that such comments are placed one indentation level to the
left of code.  Specifying
.B \-d0
lines up these comments with the code.  See the section on comment
indentation below.
.TP
.BI \-di n
Specify the indentation, in character
positions, from a declaration keyword
to the following identifier.  The default is
.B \-di16 .
.if 0 \{.TP 15
.BR \-dj , \-ndj
.B \-dj
left justifies declarations.
.B \-ndj
indents declarations the same as code.  The default is
.BR \-ndj .
.TP
.BI \-ei , \-nei
Enable (disables) special
.B else-if
processing.  If it's enabled,
.BR if "s"
following
.BR else "s"
will have the same indentation as the preceding
.B if
statement.\}
.TP
.BR \-eei , \-neei
If
.B \-eei
is specified, and extra expression indent
is applied on continuation lines
of the expression part of
.BR if (\|)
and
.BR while (\|).
These continuation lines will
be indented one extra level \(em twice instead
of just once.  This is to avoid
the confusion between the continued expression
and the statement that
follows the
.BR if (\|)
or
.BR while (\|).
Default: 
.B \-neei.
.TP
.BI \-fc1 , \-nfc1
Enables (disables) the formatting of comments that start in column 1.
Often, comments whose leading
.RB ` / '
is in column 1 have been carefully
hand formatted by the programmer.  In such cases,
\fB\-nfc1\fR
should be
used.  The default is
.BR \-fc1 .
.TP
.BI \-i n
The number of spaces for one indentation level.  The default is 4.
.TP
.BI \-ip , \-nip
Enables (disables) the indentation of parameter
declarations from the left
margin.  The default is
.B \-ip .
.TP
.BI \-l n
Maximum length of an output line with a trailing comment.
The default is 78.
.TP
.BI \-lc n
Sets the line length for block comments to
.IR n .
It defaults to being
the same as the usual line length as specified with
.BR \-l .
.TP
.BI \-lp , \-nlp
Lines up code surrounded by parenthesis
in continuation lines.  If a line
has a left paren which is not closed
on that line, then continuation lines
will be lined up to start at the character
position just after the left
parenthesis.  For example, here is how a piece
of continued code looks with
.B \-nlp
in effect:
.ne 2
.RS
.RS
.sp .5
.nf
.B "p1 = first_procedure(second_procedure(p2, p3),"
.B "                     third_procedure(p4, p5));"
.fi
.RE
.RE
.ne 5
.IP
With
.B \-lp
in effect (the default) the code looks somewhat
clearer:
.RS
.RS
.sp .5
.nf
.B "p1 = first_procedure(second_procedure(p2, p3),"
.B "                     third_procedure(p4, p5));"
.fi
.RE
.RE
.ne 5
.IP
Inserting a couple more
.SM NEWLINE
characters we get:
.RS
.RS
.sp .5
.nf
.B "p1 = first_procedure(second_procedure(p2,"
.B "                                      p3),"
.B "                     third_procedure(p4,"
.B "                                     p5));"
.fi
.RE
.RE
.TP
.B \-npro
Ignore the profile files,
.B .\|/.indent.pro
and
.BR ~/.indent.pro .
.TP
.B \-pcs , \-npcs
If true
.RB ( \-pcs )
all procedure calls will have a space inserted between
the name and the
.RB ' ( '.
The default is
.B \-npcs
.TP
.B \-psl , \-npsl
If true
.RB ( \-psl )
the names of procedures being defined are placed in
column 1 \(em their types, if any, will be
left on the previous lines.  The
default is
.BR -psl .
.TP
.BI \-sc , \-nsc
Enables (disables) the placement of asterisks
.RB (` * 's)
at the left edge of all
comments.
.TP
.BR \-sob , \-nsob
If
.B \-sob
is specified, indent will swallow optional
blank lines.  You can use this to
get rid of blank lines after declarations.  Default:
.B \-nsob
.TP
.B \-st
.B indent
takes its input from the standard input, and put its output to the
standard output.
.TP
.BI \-T " typename"
Add
.I typename
to the list of type keywords.  Names accumulate:
.B \-T
can be specified more than once.
You need to specify all the typenames that
appear in your program that are defined by
.BR typedef s
\(em nothing will
be harmed if you miss a few, but the
program won't be formatted as nicely as
it should.  This sounds like a painful thing
to have to do, but it is really a symptom of a problem in C:
.B typedef
causes a syntactic change in the
language and
.I indent
cannot find all
.BR typedef s.
.TP
.B \-troff
Causes
.B indent
to format the program for processing by
.BR troff .
It will produce a fancy
listing in much the same spirit as
.BR vgrind.
If the output file is not specified, the default is standard output,
rather than formatting in place.
The usual way to get a
.BR troff ed
listing is with the command
.RS
.IP
.BI "indent -troff " program.c " | troff -mindent"
.RE
.TP
.BR \-v , \-nv
.B \-v
turns on \*(lqverbose\*(rq mode,
.B \-nv
turns it off.  When in verbose mode,
.B indent
reports when it splits one line of input
into two or more lines of output,
and gives some size statistics at completion. The default is
.BR \-nv .
.SH "FURTHER DESCRIPTION"
.LP
You may set up your own \*(lqprofile\*(rq of defaults to
.B indent
by creating a file called
.B .indent.pro
in either your login directory or the
current directory and including
whatever switches you like.  An
.B .indent.pro
in the current directory takes
precedence over the one in your login directory.  If
.B indent
is run and a profile file exists, then
it is read to set up the program's
defaults.  Switches on the command line,
though, always override profile
switches.  The switches should be separated by
.SM SPACE\s0,
.SM TAB\s0,
or
.SM NEWLINE
characters.
.SS Comments
.TP 15
Boxed
.B indent
assumes that any comment with a dash or star
immediately after the start of
comment (that is,
.RB ` /*\-' or `/** ')
is a comment surrounded by a box of stars.
Each line of such a comment is left unchanged,
except that its indentation
may be adjusted to account for the change
in indentation of the first line
of the comment.
.br
.ne 5
.TP
Straight text
All other comments are treated as straight text.
.B indent
fits as many words (separated by
.SM SPACE\s0,
.SM TAB\s0,
or
.SM NEWLINE
characters) on a
line as possible.  Blank lines break paragraphs.
.SS Comment indentation
.LP
If a comment is on a line with code it is started
in the \*(lqcomment column\*(rq,
which is set by the
.BI \-c n
command line parameter.  Otherwise, the comment is started at
.I n
indentation levels less than where code is
currently being placed, where
.I n
is specified by the
.BI \-d n
command line parameter.  If the code on a
line extends past the comment
column, the comment starts further to the
right, and the right margin may be
automatically extended in extreme cases.
.SS Preprocessor lines
.LP
In general,
.B indent
leaves preprocessor lines alone.  The only
reformatting that it will do is to straighten
up trailing comments.  It
leaves imbedded comments alone.  Conditional compilation
.RB ( #ifdef\|.\|.\|.\|#endif )
is recognized and
.B indent
attempts to correctly
compensate for the syntactic peculiarities introduced.
.SS C syntax
.LP
.B indent
understands a substantial amount about the syntax of C, but
it has a \*(lqforgiving\*(rq parser.
It attempts to cope with the usual sorts of
incomplete and misformed syntax.
In particular, the use of macros like:
.IP
.B #define forever for(;;)
.LP
is handled properly.
.SH FILES
.PD 0
.TP 20
.B .\|/.indent.pro
profile file
.TP
.B ~/.indent.pro
profile file
.TP
.B /usr/share/lib/tmac/tmac.indent
troff macro package for 
.RB ` "indent \-troff" '
output.
.PD
.SH SEE ALSO
.BR ls (1V),
.BR troff (1)
.SH BUGS
.B indent
has even more switches than
.BR ls (1V).
.LP
A common mistake that often causes grief is typing:
.IP
.B indent *.c
.LP
to the shell in an attempt to indent all the C
programs in a directory.
This is probably a bug, not a feature.
.LP
The
.B \-bs
option splits an excessively fine hair.


usr.bin/sunwindows/sunview/defaults/lib/Indent.d
SunDefaults_Version 2
;	@(#)Indent.d	1.1	92/07/30 SMI
/Indent			""
//$Specialformat_to_defaults "indentpro_to_defaults"
//$Defaults_to_specialformat "defaults_to_indentpro"

//Indentation	"8"
; -in. Default = 8.
	$Help		"Specifies the number of spaces for one indentation level."
	$Enumeration	""
	8		""
	8/$Help		"Indent 8 spaces for each level of indentation."
	4		""
	4/$Help		"Indent 4 spaces for each level of indentation."
	
//Indentation_in_declarations	"16"
	$Help		"Indentation from the beginning of a declaration keyword to the next identifier."
; -din. Default = 16.

//Line_length	"75"
	$Help		"Maximum length of an output line."
; -ln. Default = 75.

//Case_labels	"Align_with_switch"
; -clin. Default = -cli0.
	$Enumeration
	$Help		"Controls placement of case labels relative to switch."
	Align_with_switch	""
	Align_with_switch/$Help	"Align case labels with switch."
	Indent_half_tab		""
	Indent_half_tab/$Help	"Indent case labels 1/2 tab stop from switch."

//Trailing_comments_start_in_column	"33"
	$Help		"The column in which comments other than block comments start."
; -cn. Default = 33.

//Typedefs	""
	$Help		"Defines a list of type keywords, e.g., Panel Menu Textsw."


//Mess1/$Message	""

//Mess4/$Message	"    ~~~~~~ declarations ~~~~~~"

//Parameter_declarations	"Indented"
; -ip, -nip. Default is -ip.
        $Enumeration	""
        $Help           "Specifies the indentation of parameter declarations from the left margin."
	Indented		""
        Indented/$Help           "Parameter declarations are indented one level from the left margin."
	In_column_1		""
	In_column_1/$Help           "Parameter declarations start in column 1."

//Multiple_declarations		"On_same_lines"
; -bc, -nbc. Default = -nbc.
        $Enumeration	""
        $Help           "Enables (disables) multiple declarations of the same type on the same line."
	On_same_lines		""
        On_same_lines/$Help           "Multiple declarations of same type can apear on same line, e.g. int x, y, z;"
	On_separate_lines		""
	On_separate_lines/$Help           "Force multiple declarations onto separate lines."


//Return_type_of_procedure		"On_previous_line"
; -psl, -npsl. Default = -psl.
        $Enumeration	""
        $Help           "Controls placement of the return type of a procedure relative to the procedure name."
	On_previous_line		""
        On_previous_line/$Help           "Place return type of procedure on previous line from name of procedure."
	On_same_line		""
	On_same_line/$Help           "Place return type of procedure on same line as name of procedure."


//Mess5/$Message	""
//Mess6/$Message	"    ~~~~~~ block comments ~~~~~~"

//Comment_delimiters	"On_separate_line"
; -cdb, -ncdb, Default = -cdb.
        $Enumeration	""
        $Help           "Enables/disables the placement of block comment delimiters (/* and */) on separate lines."
	On_separate_line		""
        On_separate_line/$Help           "Comment delimiters for block comments will always be on separate lines."
	On_same_line		""
	On_same_line/$Help           "Comment delimiters for block comments will be place on the same line as the comment."

//Asterisk_at_left_edge	"Yes"
; -sc, -nsc, Default = -sc.
        $Enumeration	""
        $Help           "Enables (disables) the placement of *'s at the left edge of each line of a block comment."
	Yes		""
	Yes/$Help	"A * will appear at the left edge of each line of a block comment."
	No
	No/$Help	"A * will NOT appear at the left edge of each line of a block comment."

//Indent_block_comments		"Same_as_commented_code"
; -dn. Default = -d0.
        $Enumeration	""
        $Help           "Controls placement of block comments."
	Same_as_commented_code		""
	Same_as_commented_code/$Help	"Block comments line up with code."
	Indent_LEFT_one_level
	Indent_LEFT_one_level/$Help	"BLock comments are placed one indentation level to the LEFT of the code."

//Format_comments_in_column_1	"Yes"
; -fc1, -nfc1. Default = -fc1.
	$Help	"Enables (disables) the formatting of comments that start in column 1."
	$Enumeration
	Yes
	Yes/$Help	"Comments that start in column 1 are formatted."
	No
	No/$Help	"Comments that start in column 1 are left alone."


//Mess7/$Message	""
//Mess8/$Message	"    ~~~~~~ blank lines ~~~~~~"

//Blank_line_after_proc_body	 "Yes"
; -bap, nbap. Default = -bap.
	$Help	"Controls insertion of blank lines following procedure bodies."
	$Enumeration
	Yes
	Yes/$Help	"Blank line inserted after every procedure body."
	No
	No/$Help	"Blank line is NOT inserted after procedure body."

//Blank_line_after_decl_block	 "Yes"
; -bad, -nbad. Default = -bad.
	$Help	"Controls insertion of blank lines following block of declarations."
	$Enumeration
	Yes
	Yes/$Help	"Blank line inserted after every block of declarations."
	No
	No/$Help	"Blank line is NOT inserted after block of declarations."

//Blank_line_before_block_comment	 "No"
; -bbb,-nbbb. Default = -nbbb.
	$Help	"Controls insertion of blank lines before block comments."
	$Enumeration
	Yes
	Yes/$Help	"Blank line inserted before every block commented."
	No
	No/$Help	"Blank line is NOT inserted before block comments."

//Swallow_optional_blank_lines	"No"
; -sob, -nsob. Default = -nsob.
	$Help	"Enables (disables) swallowing of optional blank lines, e.g., after declarations."
	$Enumeration
	Yes
	Yes/$Help	"Indent will swallow optional blank lines, e.g., after declarations."
	No
	No/$Help	"Blank lines are not swallowed."


//Mess9/$Message	""
//Mess10/$Message	"    ~~~~~~  braces ~~~~~~"

//Left_brace	"On_same_line"
; -br, -bl. Default = -br.
	$Help	"Controls placement of {'s in compound statements."
	$Enumeration	""
	On_same_line		""
        On_same_line/$Help	"Left brace appears on same line as preceding code, e.g., if (...) {"
	On_separate_line	""
	On_separate_line/$Help           "Left brace always appears on a line by itself."

//Right_brace_and_else	"On_same_line"
; -ce, -nce. Default = -ce.
        $Enumeration	""
        $Help           "Enables (disables) forcing else's to cuddle up to the immediately preceding }"
	On_same_line		""
        On_same_line/$Help           "else appears on same line as preceding }, e.g. } else {"
	On_separate_line		""
	On_separate_line/$Help		"else appears on line after }"


//Mess11/$Message	""
//Mess12/$Message	"    ~~~~~~ miscellaneous  ~~~~~~"

	
//Space_after_proc_call		"No"
; -pcs, -npcs. Default = -npcs.
        $Enumeration	""
        $Help           "Controls insertion of space between procedure name and '(' in procedure call."
	Yes		""
        Yes/$Help           "Insert space between procedure name and '(', e.g., foo (args)"
	No		""
	No/$Help		"Don't insert space between procedure name and '(', e.g., foo(args)"

//Align_parenthesized_code		"Yes"
; -lp, -nlp. Default = -lp.
        $Enumeration	""
        $Help           "Controls formatting of continuation lines."
	Yes		""
        Yes/$Help           "When inside parentheses, align continuation lines at position of '(' + 1"
	No		""
	No/$Help		"Being inside parentheses does not affect formatting of continuation lines."


//Other_options		""
	$Help		"A string of options to be passed in to indent, e.g. -nbs -troff."
